guides/v2.3/graphql/reference/paypal.md

```diff
@@ -0,0 +1,231 @@
+            urls: {
+                return_url: "http://magento.test/paypal/express/return/"
+                cancel_url: "http://magento.test/paypal/express/cancel/"
+            }
+        }
+    )
+    {
+        token
+        paypal_urls{
+            start
+            edit
+        }
+    }
+}
+```
+
+**Response**
+
+```json
+{
+  "data": {
+    "createPaypalExpressToken": {
+      "token": "<PayPal_Token>"
+      "paypal_urls": {
+        "start": "https://www.sandbox.paypal.com/checkoutnow?token=<PayPal_Token>"
+        "edit": "https://www.sandbox.paypal.com/cgi-bin/webscr?cmd=_express-checkout&useraction=continue&token=<PayPal_Token>"
+      }
+    }
+  }
+}
+```
+
+### Set the payment method
+
+Magento GraphQL supports the `paypal_express` and `paypal_payflow` payment methods.
+
+**Request**
+
+```text
+mutation {
+  setPaymentMethodOnCart(input: {
+    cart_id: "rMQdWEecBZr4SVWZwj2AF6y0dNCKQ8uH"
+    payment_method: {
+        code: "paypal_express"
+        additional_data: {
+            paypal_express: {
+                payer_id: "<PayPal_PayerID>"
+                token: "<PayPal_Token>"
+            }
+        }
+      }
+  }) {
+    cart {
+      selected_payment_method {
+        code
+        title
+      }
+    }
+  }
+}
+```
+
+**Response**
+
+```json
+{
+  "data": {
+    "setPaymentMethodOnCart": {
+      "cart": {
+        "selected_payment_method": {
+          "code": "paypal_express",
+          "title": "PayPal Express Checkout"
+        }
+      }
+    }
+  }
+}
+```
+
+### Place the order
+
+**Request**
+
+``` text
+mutation {
+  placeOrder(
+    input: {
+      cart_id: "rMQdWEecBZr4SVWZwj2AF6y0dNCKQ8uH"
+    }
+  ) {
+    order {
+      order_id
+    }
+  }
+}
+```
+
+**Response**
+
+```json
+{
+  "data": {
+    "placeOrder": {
+      "order": {
+        "order_id": "000000006"
+      }
+    }
+  }
+}
+```
+
+## Input attributes
+
+### PaypalExpressTokenInput {#PaypalExpressTokenInput}
+
+The `PaypalExpressTokenInput` object defines the attributes required to receive a payment token from PayPal
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`cart_id` | String! | The unique ID that identifies the customer's cart
+`code` | String! | Payment method code
+`express_button`: | Boolean | Indicates whether the buyer selected the PayPal Express checkout button. The default value is `false`
+`urls` | [`PaypalExpressUrlsInput`](#PaypalExpressUrlsInput) | Defines a set of URLs to redirect to in response to the token request
+`use_paypal_credit` | Boolean | Indicates whether the buyer clicked the Paypal credit button. The default value is `false`
+
+### PaypalExpressUrlsInput {#PaypalExpressUrlsInput}
+
+The `PaypalExpressUrlsInput` object contains a set of URLs that PayPal uses to respond to a token request.
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`cancel_url` | String! | The URL of the original page on your website where the buyer initially chose PayPal as a payment type
```

Comment:
Just wanting to confirm that this description is correct for `cancel_url`? I would expect to see something about canceling...but maybe this is a Web term I'm not familiar with.

Reply:
Clarified.